core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/SystemSettingsCredentialsProvider.java

```diff
@@ -43,12 +46,28 @@
  */
 @SdkInternalApi
 public abstract class SystemSettingsCredentialsProvider implements AwsCredentialsProvider {
+
+    private AwsCredentialsProvider credentialsProvider;
+
     @Override
     public AwsCredentials resolveCredentials() {
+
+        if (credentialsProvider == null) {
+            resolveProvider();
+        }
```

Comment:
Any reason not to just do this on creation, instead of first use?

Reply:
We could do this but it would force us to remove some tests we have that instantiate an environment variable credential provider.

Reply:
Why would we have to remove the tests? This is a lot of complexity, and I don't see the benefit.